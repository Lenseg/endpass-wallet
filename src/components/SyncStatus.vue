<template>
	<div>
		<p class="heading">Block number</p>
		<div class="tags has-addons">
		  <span class="tag">{{blockNumber}}</span>
			<span class="tag is-warning" v-if="isSyncing">Syncing</span>
			<span class="tag is-success" v-else>Synced</span>
		</div>
	</div>
</template>
<script>
import { mapState } from 'vuex';
export default {
	data() {
		return {

		}
	},
	computed: {
		...mapState({
	  	blockNumber: state => state.web3.blockNumber,
	  	isSyncing: state => state.web3.isSyncing,
	  })
	}
}
</script>
<style>
	
</style>