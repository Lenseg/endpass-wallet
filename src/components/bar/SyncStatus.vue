<template>
	<div>
		<p class="heading">Block number</p>
		<div class="tags has-addons">
		  <span class="tag">{{blockNumber}}</span>
			<span class="tag" :class="'is-'+appStatus"></span>
		</div>
	</div>
</template>
<script>
import { mapState, mapGetters } from 'vuex';
export default {
	data() {
		return {

		}
	},
	computed: {
		...mapState({
	  	blockNumber: state => state.web3.blockNumber,
	  	account: state => state.accounts.activeAccount,
	  	balance: state => state.accounts.balance
	  }),
		...mapGetters('connectionStatus', [
			'appStatus'
		])
	}
}
</script>
<style>

</style>
