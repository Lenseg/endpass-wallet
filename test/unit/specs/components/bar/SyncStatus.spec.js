import { shallow, createLocalVue } from '@vue/test-utils';

import Vuex from 'vuex';
const localVue = createLocalVue();

localVue.use(Vuex);

import SyncStatus from '@/components/bar/SyncStatus';

describe('SyncStatus', () => {
  let wrapper;
  beforeEach(() => {
    const store = new Vuex.Store({
      modules: {
        web3: {
          namespaced: true,
          state: {
            blockNumber: 1,
          },
        },
        connectionStatus: {
          namespaced: true,
          getters: {
            appStatus: () => 'failed',
          },
        },
      },
    });
    wrapper = shallow(SyncStatus, {
      localVue,
      store,
    });
  });
  describe('render', () => {
    it('should be a Vue component', () => {
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should match snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render data from vuex', () => {
      const tag = wrapper.find('.tag');
      expect(tag.element.title).toBe('synced to block 1');
      expect(tag.classes()).toContain('is-danger');
      expect(tag.text()).toContain('failed');
    });
  });

  describe('computed', () => {
    it('should return correct statusClass', () => {
      expect(wrapper.vm.statusClass).toBe('is-danger');
      wrapper.setComputed({ appStatus: 'syncing' });
      expect(wrapper.vm.statusClass).toBe('is-warning');
      wrapper.setComputed({ appStatus: 'ready' });
      expect(wrapper.vm.statusClass).toBe('is-success');
    });
  });
});
