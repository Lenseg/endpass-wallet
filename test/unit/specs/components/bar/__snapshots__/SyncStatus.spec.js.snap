// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncStatus render should match snapshot 1`] = `
<div
  class="sync-status"
>
  <span
    class="tag is-danger"
    title="synced to block 1"
  >
    failed
  </span>
</div>
`;
